### PR TITLE
GLM-5.2 FP8 compatibility on MUSA

### PR DESCRIPTION
**Pull request by Joey-gvwal**
## Summary

Enable GLM-5.2-FP8 sparse MLA inference on vllm-musa for the vLLM 0.24 codebase.
This series wires the GLM DSA sparse-MLA backend into the MUSA platform, routes the
sparse FlashMLA implementation to MUSA-native ops, replaces the hanging sparse
attention kernel path with the SGLang TileLang kernel, and fixes two query-buffer
initialization bugs that caused precision/output corruption.

The final validated serving mode is two-node TP8 + PP2 + EP, eager mode,
`VLLM_MOE_USE_DEEP_GEMM=0`, model `/home/dist/models/zai-org/GLM-5.2-FP8`.

## Modifications

- Register `FLASHMLA_SPARSE` for the MUSA platform and add the sparse MLA backend
  to the MUSA v1 attention backend selection path.
- Route sparse FlashMLA metadata and attention operations through the MUSA
  `flash_mla` implementation instead of CUDA-only upstream paths.
- Add the TileLang sparse MLA kernel implementation and use
  `musa_sparse_attention_fwd_kernel_v1` with the corrected KV shared-layout
  annotations.
- Keep GLM-5.2 serving on the stable path by using expert parallel and disabling
  DeepGEMM MoE at runtime for this model path.
- Add build patch
  `vllm_musa/patches/series/0085-MUSA-vllm.v1.attention.backends.mla.flashmla_sparse.patch`
  so a clean vllm-musa install reproduces the GLM sparse MLA query-buffer fixes:
  `new_empty` -> `new_zeros` for BF16 sparse prefill head padding, and
  `concat_mla_q` with a reused workspace -> fresh full-write `torch.cat`.

Commits covered:

- `434995e2` MUSA: wire FLASHMLA_SPARSE backend into the MLA selector for DSA models
- `ae24f1e4` MUSA: route sparse FlashMLA ops to the MUSA flash_mla lib for DSA inference
- `2855da2b` MUSA: use SGLang num_stages=0 TileLang DSA kernel for sparse-MLA
- `3ef7fc4e` MUSA: use SGLang musa_sparse_attention_fwd_kernel_v1
- `ee10f35c` MUSA: fix GLM sparse MLA query buffer initialization

### Two-node deployment command

Common runtime environment:

```bash
export GLOO_SOCKET_IFNAME=bond0
export VLLM_WORKER_MULTIPROC_METHOD=spawn
export PYTHONUNBUFFERED=1
export VLLM_DISABLE_COMPILE_CACHE=1
export VLLM_MOE_USE_DEEP_GEMM=0
```

Head/API node:

```bash
vllm serve /home/dist/models/zai-org/GLM-5.2-FP8 \
  --tensor-parallel-size 8 \
  --pipeline-parallel-size 2 \
  --enable-expert-parallel \
  --nnodes 2 \
  --node-rank 0 \
  --master-addr $Master_IP \
  --master-port 29500 \
  --trust-remote-code \
  --served-model-name glm52 \
  --max-model-len 16384 \
  --tool-call-parser glm47 \
  --enable-auto-tool-choice \
  --reasoning-parser glm45 \
  --enforce-eager \
  --host 127.0.0.1 \
  --port 8012
```

Worker node:

```bash
vllm serve /home/dist/models/zai-org/GLM-5.2-FP8 \
  --tensor-parallel-size 8 \
  --pipeline-parallel-size 2 \
  --enable-expert-parallel \
  --nnodes 2 \
  --node-rank 1 \
  --master-addr $Master_IP \
  --master-port 29500 \
  --trust-remote-code \
  --served-model-name glm52 \
  --max-model-len 16384 \
  --tool-call-parser glm47 \
  --enable-auto-tool-choice \
  --reasoning-parser glm45 \
  --enforce-eager \
  --headless
```

Health/model checks:

```bash
curl -sS http://127.0.0.1:8012/health
curl -sS http://127.0.0.1:8012/v1/models
```

### Output precision anomalies and fix effects

| Before fix | Root cause | Fix | Effect after fix |
|---|---|---|---|
| GLM-5.2 produced deterministic garbage text such as CJK/noisy output. Debug showed finite pre-absorption q, then NaN in sparse MLA kernel input. | BF16 sparse prefill padded TP8 heads from 8 to 64 with `q.new_empty`, leaving uninitialized padded heads. The sparse kernel consumed the 64-head tile and propagated NaN into real heads. | Change BF16 sparse prefill padding to `q.new_zeros`. | Padding region became zero and the output changed, proving the active path was fixed, but output was still degenerate. |
| After zeroing padded heads, output changed to another degenerate pattern such as repeated `in` / `responsibleComments`. Debug showed padded heads were fixed, but real heads still contained NaN/Inf after query concatenation. | `ops.concat_mla_q` wrote into a reused `q_concat_buffer` and did not fully overwrite it, so stale NaN/garbage survived even though `ql_nope` and `q_pe` inputs were finite. | Replace reused buffer + `ops.concat_mla_q` with fresh full-write `torch.cat([ql_nope, q_pe], dim=-1)`. | Correctness gate passed. France prompt identified Paris, arithmetic prompt returned `42`, and exact phrase prompt returned `GLM correctness passed.` |
| TP8 MoE path had a separate block-scale shard issue and could corrupt model semantics. | MoE TP8 scale rows were not reliable for this GLM-5.2 path. | Run with `--enable-expert-parallel` and `VLLM_MOE_USE_DEEP_GEMM=0` to use the TRITON FP8 MoE path. | Serving and correctness checks completed on TP8 + PP2 + EP. |

### Correctness checks

Fresh two-node serve after both sparse MLA query-buffer fixes:

- `What is the capital of France?` -> reasoning identifies Paris.
- `Answer with only the integer: 17 plus 25.` -> content `42`.
- `Reply exactly with this text and nothing else: GLM correctness passed.` -> content `GLM correctness passed.`

### CODE REVIEW

**Comment by Joey-gvwal** [comment]
The latest commit updated the vLLM pin. I'll revert it and submit a new commit instead.
